lib/cluster-providers/generic-cluster-provider.ts

```diff
@@ -98,7 +159,13 @@ export class ClusterBuilder {
 export class GenericClusterProvider implements ClusterProvider {
 
     constructor(readonly props: GenericClusterProviderProps) {
-        assert(!(props.managedNodeGroups && props.managedNodeGroups.length > 0 
+
+        validateConstraints(props, genericClusterPropsContraints, GenericClusterProvider.name);
```

Comment:
Move line 163 - 168 to a separate method `validateInput` (similar to what we have in `EKSBlueprintStack`)

Reply:
That is a good call, much cleaner now.